### ISSUE TITLE
Support screen reader on Linux

### ISSUE THREAD

**Issue by nishant09** [microsoft/vscode#19522]
- VSCode Version: 1.5.3
- OS Version: Ubuntu 14.04

Steps to Reproduce:

1. Turn On screen reader on Ubuntu (http://askubuntu.com/questions/278693/how-do-i-stop-orca-screen-reader)
2. Open VS code
3. Naviagate through items with Tab key

Expected behavior:

Screen reader should read items. (as how it reads on Mac & Windows)

Actual behavior:

It doesn't read anything. 

Is this a known issue ? Do you know any alternative screen reading tool for ubuntu ?

Any help will be appreciated !

**Comment by Tyriar** [comment]
@bpasero is it an electron issue that screen readers only work on Windows/Mac?

**Comment by bpasero** [comment]
Yes, I am quite sure we never supported a screen reader on Linux (Chromium does not as far as I know).

**Comment by miksuh** [comment]
It is about time you start to support Orca screen reader on Linux. There is now applications like Skype which use Electron, that means that e.g Skype for Linux is now completely unusable if you are blind like I am. So because of you blind users can not use Skype on Linux. And it is not juse Skype, all Electron apps are 100% unusable and so is chromium browser. This is simply discriminating blind and visually impaired users. It is not accessptable that blind users can not use e.g Skype just because chromium and electron developers do not seem to care about accessibility on Linux at all. And do not say that I should be more polite. If you do not care to support accessibility and Orca screen reader on Linux, then I sure do not care to be polite. You have had years of time to make chromium and electron accessible with Orca, and there has been several people asking you to do that. But absoletely nothing has happened. I really do not understand what is wrong with chromium and electron developers. Mozilla has made Firefox work just fine with Orca, why chromium developers do not care to do the same?

**Comment by tcikoritys** [comment]
electron/electron#11544

**Comment by webczat** [comment]
works already more or less on master/insider builds containing electron 7. There are issues and I am not sure if they are common to linux and windows or linux specific. Also this is not the newest chromium so that there are both screenreader and chromium issues, so hard to say at first glance. But it is now definitely usable, and bugs are more of being an annoyance than preventing usage of a feature.

**Comment by isidorn** [comment]
@webczat  Glad to hear that. Can you please file those "annoyance" issues as seperate issues on GitHub? Also please note of the issues already tagged with accessibility, just for reference https://github.com/microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility

**Comment by jvesouza** [comment]
To be honest I am quite impressed with the accessibility using the orca screen reader, taking into account that until recently there was no support at all.

I believe that as soon as there is an update to a more current version of chromium some problems will disappear.

**Comment by webczat** [comment]
that is actually why I am reluctant to report most of these annoyance issues now, because many of them are chromium upstream or orca issues and some of them could already have been fixed in newer versions of chromium. So things would clear out more in the coming months. There are likely some issues that definitely look vscode, including some that may be platform independent.

**Comment by jvesouza** [comment]
Actually I am using orca compiled from master.

**Comment by webczat** [comment]
me too.

**Comment by isidorn** [comment]
Well if those issues are already reported and fixed by chromium than us updating to a newer electron always helps. Though make sure to file issues to Chromium also for problems which they are not aware of. Thank you

**Comment by isidorn** [comment]
@jvesouza did you have to pass the special flag --force-renderer-accessibility to enable this? Or the screen reader works out of the box on linux?
Asking because if the flag is required we could automatically pass it from VS Code.
Also we should potentially update our accessibility docs to mention that we support Orca on linux https://code.visualstudio.com/docs/editor/accessibility#_linux

**Comment by webczat** [comment]
note that the chromium's linux accessibility work is not yet officially complete. and electron-7 bases off on chromium 78. So I am not sure if flags like --force-renderer-accessibility have to be passed automatically by vscode, or it's enough to wait till the work is completed. That said, the flag is required, and also an environment variable ACCESSIBILITY_ENABLED set to 1. Without that nothing will work.

**Comment by isidorn** [comment]
Ok, that sounds like they are working on it. Great that you pointed it out so people can use those flags as a workaround for now.
I have emailed chrome to get more details, will keep you posted here if I learn something more.

**Comment by jvesouza** [comment]
Yes, I have to pass the --force-renderer-accessibility flag when invoking vscode.
However I don't hav an environment variable ACCESSIBILITY_ENABLED created in my environment.
What distro are you using?
Are you using gnome?

Actually I am using arch linux with gnome.

**Comment by webczat** [comment]
me too, and in many cases not setting this variable results in the vscode not being accessible even after passing the flag.

**Comment by isidorn** [comment]
We actually suppor tnow Orca screen reader on Linux. More details can be found here 
https://github.com/microsoft/vscode/issues/90446

In short: `"editor.accessibilitySupport": "on"` has to be manually set by users and all should nicely work with latest VS Code insiders and the latest Orca out of master.
I plan to update our accessibility docs to mention this.
Thus closing this one.
